### DISCUSSION TITLE
skynet在容器，k8s，中运行

### DISCUSSION THREAD

**Discussion by kakaximark** [cloudwu/skynet#1489] [General]
如何保证skynet在容器，或者k8s环境中稳定运行?

**Comment by kakaximark** [comment]
没人做相同的项目吗？

**Comment by hanxi** [comment]
不知道你说的稳定运行是啥意思。如果是指如何在容器中保持后台模式运行 skynet 的话，可以在启动 skynet 后加入一行 tail 指令。

```shell
./skynet examples/config
tail -F /dev/null
```

**Reply by chrXiTer** [reply]
我一般是加一个死循环命令，循环体里面 sleep，不知道哪个更好。
不过这不是skynet的问题，是容器的问题。

**Comment by cloudwu** [comment]
不能在 config 中配置后台运行？

**Reply by kakaximark** [reply]
我的意思是，比如在lua游戏项目中，同一个pod中我启动多个游戏容器，做负载，这样好像是没有办法实现的，因为在内存级别来说，动态内存，是不能共享的，

**Reply by JieTrancender** [reply]
1. 游戏分服就是一种负载。
2. n个游戏服进程起m个服是一种负载。
3. n个玩家代理进程起m个玩家服务是一种负载。

**Reply by kakaximark** [reply]
没错，但是目前主要是想用到k8s，pod的特性来实现，，但是使用pod启动多个容器的过程，好像是走不通，所以有疑惑，

**Reply by hanxi** [reply]
这个跟自己游戏架构强相关的，比如某些节点是唯一的如配置中心，某些节点是能动态删减的如战斗计算服务,可以参考 k8s 里配置 mysql 主从的方案，加节点和删节点前后都需要做额外的事情的，需要自己写触发事件的。先实现手动扩容和缩容，再实现自动触发。

**Reply by kakaximark** [reply]
好的，感谢
